ldm/dream/server.py

```diff
@@ -46,50 +47,70 @@ def do_POST(self):
         height = int(post_data['height'])
         cfgscale = float(post_data['cfgscale'])
         gfpgan_strength = float(post_data['gfpgan_strength'])
-        seed = None if int(post_data['seed']) == -1 else int(post_data['seed'])
+        seed = self.model.seed if int(post_data['seed']) == -1 else int(post_data['seed'])
 
         print(f"Request to generate with prompt: {prompt}")
 
-        outputs = []
+        def image_done(image, seed):
+            config = post_data.copy() # Shallow copy
+            config['initimg'] = ''
+
+            # Write PNGs
+            pngwriter = PngWriter("./outputs/img-samples/", config['prompt'])
+            pngwriter.write_image(image, seed)
+
+            # Append post_data to log
+            with open("./outputs/img-samples/dream_web_log.txt", "a") as log:
+                for file_path, _ in pngwriter.files_written:
+                    log.write(f"{file_path}: {json.dumps(config)}\n")
+
+            self.wfile.write(bytes(json.dumps(
+                {'event':'result', 'files':pngwriter.files_written, 'config':config}
+            ) + '\n',"utf-8"))
+
+        # TODO: refactor PngWriter:
+        # it doesn't need to know if batch_size > 1, just if this is _part of a batch_
+        step_writer = PngWriter('./outputs/intermediates/', prompt, 2)
+        def image_progress(sample, step):
+            images = self.model._samples_to_images(sample)
```

Comment:
This step should be optional, as synchronizing the GPU data with the CPU can slow down generation (done here: https://github.com/lstein/stable-diffusion/blob/main/ldm/simplet2i.py#L509).

Reply:
Ah, fair point, I'll add an off-by-default checkbox. Just a sec...

Reply:
Actually, I've just going backed out my commit and will submit that part as its own PR (including the checkbox).